internal/commands/list_registries_test.go

```diff
@@ -0,0 +1,96 @@
+package commands_test
+
+import (
+	"bytes"
+	"testing"
+
+	h "github.com/buildpacks/pack/testhelpers"
+
+	"github.com/buildpacks/pack/internal/commands"
+
+	"github.com/heroku/color"
+	"github.com/sclevine/spec"
+	"github.com/sclevine/spec/report"
+	"github.com/spf13/cobra"
+
+	"github.com/buildpacks/pack/internal/config"
+	ilogging "github.com/buildpacks/pack/internal/logging"
+	"github.com/buildpacks/pack/logging"
+)
+
+func TestListRegistries(t *testing.T) {
+	color.Disable(true)
+	defer color.Disable(false)
+
+	spec.Run(t, "Commands", testListRegistriesCommand, spec.Parallel(), spec.Report(report.Terminal{}))
+}
+
+func testListRegistriesCommand(t *testing.T, when spec.G, it spec.S) {
+	var (
+		command *cobra.Command
+		logger  logging.Logger
+		outBuf  bytes.Buffer
+		cfg     config.Config
+	)
+
+	it.Before(func() {
+		logger = ilogging.NewLogWithWriters(&outBuf, &outBuf)
+		cfg = config.Config{
+			DefaultRegistryName: "private registry",
+			Registries: []config.Registry{
+				{
+					Name: "public registry",
+					Type: "github",
+					URL:  "https://github.com/buildpacks/public-registry",
+				},
+				{
+					Name: "private registry",
+					Type: "github",
+					URL:  "https://github.com/buildpacks/private-registry",
+				},
+				{
+					Name: "personal registry",
+					Type: "github",
+					URL:  "https://github.com/buildpacks/personal-registry",
+				},
+			},
+		}
+		command = commands.ListRegistries(logger, cfg)
+	})
+
+	when("#ListRegistries", func() {
+		it("should list all registries", func() {
+			h.AssertNil(t, command.Execute())
+
+			h.AssertContains(t, outBuf.String(), "public registry")
+			h.AssertContains(t, outBuf.String(), "* private registry")
+			h.AssertContains(t, outBuf.String(), "personal registry")
+		})
+
+		it("should list registries in verbose mode", func() {
+			logger := ilogging.NewLogWithWriters(&outBuf, &outBuf, ilogging.WithVerbose())
+			command = commands.ListRegistries(logger, cfg)
+
+			h.AssertNil(t, command.Execute())
+
+			h.AssertContains(t, outBuf.String(), "github")
+
+			h.AssertContains(t, outBuf.String(), "public registry")
+			h.AssertContains(t, outBuf.String(), "https://github.com/buildpacks/public-registry")
+
+			h.AssertContains(t, outBuf.String(), "private registry")
+			h.AssertContains(t, outBuf.String(), "https://github.com/buildpacks/private-registry")
+
+			h.AssertContains(t, outBuf.String(), "personal registry")
+			h.AssertContains(t, outBuf.String(), "https://github.com/buildpacks/personal-registry")
+		})
+
+		it("should list nothing when NO registries exist", func() {
+			cfg = config.Config{}
+			command = commands.ListRegistries(logger, cfg)
+
+			h.AssertNil(t, command.Execute())
+			h.AssertEq(t, outBuf.String(), "")
+		})
```

Comment:
What do you think about logging something like "No registries have been added yet. You can do that by `pack add-buildpacks-registry`"? I'd be confused if I ran this and nothing happened.

Reply:
@dfreilich I could go either way.  Part of me is inclined to think that if the user is confused they could look at the command's help menu to better understand its behavior.  I'll let @jromero be the tie breaker :smiley:

Reply:
@jromero do you have any strong thoughts on this?